pkg/security/controller/namespace_scc_allocation_controller.go

```diff
@@ -0,0 +1,339 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	coreapi "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/controller"
+
+	securityv1 "github.com/openshift/api/security/v1"
+	securityv1client "github.com/openshift/client-go/security/clientset/versioned/typed/security/v1"
+	"github.com/openshift/origin/pkg/security"
+	"github.com/openshift/origin/pkg/security/mcs"
+	"github.com/openshift/origin/pkg/security/uid"
+	"github.com/openshift/origin/pkg/security/uidallocator"
+)
+
+const (
+	controllerName = "namespace-security-allocation-controller"
+	rangeName      = "scc-uid"
+)
+
+// SCCAllocationController allocates uids/labels for namespaces
+type NamespaceSCCAllocationController struct {
+	requiredUIDRange          *uid.Range
+	mcsAllocator              MCSAllocationFunc
+	nsLister                  corev1listers.NamespaceLister
+	nsListerSynced            cache.InformerSynced
+	currentUIDRangeAllocation *securityv1.RangeAllocation
+
+	namespaceClient       corev1client.NamespaceInterface
+	rangeAllocationClient securityv1client.RangeAllocationsGetter
+
+	queue workqueue.RateLimitingInterface
+}
+
+func NewNamespaceSCCAllocationController(
+	namespaceInformer corev1informers.NamespaceInformer,
+	client corev1client.NamespaceInterface,
+	rangeAllocationClient securityv1client.RangeAllocationsGetter,
+	requiredUIDRange *uid.Range,
+	mcs MCSAllocationFunc,
+) *NamespaceSCCAllocationController {
+	c := &NamespaceSCCAllocationController{
+		requiredUIDRange:      requiredUIDRange,
+		mcsAllocator:          mcs,
+		namespaceClient:       client,
+		rangeAllocationClient: rangeAllocationClient,
+		nsLister:              namespaceInformer.Lister(),
+		nsListerSynced:        namespaceInformer.Informer().HasSynced,
+		queue:                 workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), controllerName),
+	}
+
+	namespaceInformer.Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: c.enqueueNamespace,
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				c.enqueueNamespace(newObj)
+			},
+		},
+		10*time.Minute,
+	)
+	return c
+}
+
+// Run starts the workers for this controller.
+func (c *NamespaceSCCAllocationController) Run(stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	defer glog.V(1).Infof("Shutting down")
+
+	// Wait for the stores to fill
+	if !controller.WaitForCacheSync(controllerName, stopCh, c.nsListerSynced) {
+		return
+	}
+
+	glog.V(1).Infof("Repairing SCC UID Allocations")
+	if err := c.WaitForRepair(stopCh); err != nil {
+		// this is consistent with previous behavior
+		glog.Fatal(err)
+	}
+	glog.V(1).Infof("Repair complete")
```

Comment:
What is the benefit of showing these messages on V1 level?

Reply:
> What is the benefit of showing these messages on V1 level?

One time stage completion messages?  Seems like exactly what you want for V(1) along with detailed failure reasons.